hydra-node/test/Hydra/API/RestServerSpec.hs

```diff
@@ -5,15 +5,36 @@ module Hydra.API.RestServerSpec where
 import Hydra.Prelude
 import Test.Hydra.Prelude
 
+import Data.Aeson.Lens (key)
 import Hydra.API.RestServer (DraftCommitTxRequest, DraftCommitTxResponse)
 import Hydra.Chain.Direct.State ()
+import Hydra.JSONSchema (SpecificationSelector, prop_specIsComplete, prop_validateJSONSchema)
 import Hydra.Ledger.Cardano (Tx)
 import Test.Aeson.GenericSpecs (roundtripAndGoldenSpecs)
+import Test.QuickCheck.Property (conjoin, property, withMaxSuccess)
 
 spec :: Spec
 spec = parallel $ do
   roundtripAndGoldenSpecs
     (Proxy @(ReasonablySized (DraftCommitTxResponse Tx)))
 
   roundtripAndGoldenSpecs
-    (Proxy @(ReasonablySized (DraftCommitTxRequest Tx)))
+    (Proxy @(ReasonablySized DraftCommitTxRequest))
+
+  prop "Validate /commit publish api schema" $
+    property $
+      withMaxSuccess 1 $ do
+        conjoin
+          [ prop_validateJSONSchema @DraftCommitTxRequest "api" (key "components" . key "messages")
+          ]
+
+  prop "Validate /commit subscribe api schema" $
+    property $
+      withMaxSuccess 1 $ do
+        conjoin
+          [ prop_validateJSONSchema @(DraftCommitTxResponse Tx) "api" (key "channels" . key "/commit" . key "subscribe")
+          , prop_specIsComplete @(DraftCommitTxResponse Tx) "api" apiSpecificationSelector
```

Comment:
This is a bit pointless as `DraftCommitTxResponse` is a product type and has not multiple variants which could be checked for completeness. However, the `/commit` endpoint should have multiple responses, also in the api.yaml [here](https://github.com/input-output-hk/hydra/pull/903/files#diff-f20909c04bf600ba813ac22890c66583bc03b69c1d210fc2b7b3ff30c8d9fb6eR120-R121)

Reply:
Should: Specify the possible errors in the `/commit` subscribe endpoint. It should be `oneOf` the nominal response and the possible error responses we send. Looking at https://github.com/input-output-hk/hydra/pull/903/files#diff-f1fbd8fea2154d2538329d3e931106afa899ed70716d79fc58b84ed2a0406b42R387-R390, I would expect at least the 400 responses to be documented in the api.